### ISSUE TITLE
TextField select all text on focus

### ISSUE THREAD

**Issue by pordonj** [gui-cs/Terminal.Gui#1474]
If `TextField` gets focus via keyboard: select all text on receiving focus. 

If Set focus by mouse, deselect and positioned insertion point on point where was clicked. 

We should ensure there are sufficient parallelizable unit tests for this too. 

All tests must pass before committing.

**Comment by tig** [comment]
I agree this should be the default behavior. This is how single-line text boxes in dialogs in Windows works.

**Comment by BDisp** [comment]
> I agree this should be the default behavior. This is how single-line text boxes in dialogs in Windows works.

Maybe with the exception if there is some text already selected?

**Comment by tig** [comment]
> > I agree this should be the default behavior. This is how single-line text boxes in dialogs in Windows works.
> 
> Maybe with the exception if there is some text already selected?

On Windows, the previous selection is ignored. So I don't think so.

**Comment by BDisp** [comment]
> On Windows, the previous selection is ignored. So I don't think so.

No it doesn't ignore. See bellow. What we can do is a flag `SelectAllTextOnFocus` where the user can choose select all text on focus.

![selected-text-preserved](https://user-images.githubusercontent.com/13117724/182392701-f1d56bc8-01cf-40d4-ad75-527d00d2d761.gif)

**Comment by tig** [comment]
Is that a Visual Studio dialog?

All my tests of dialogs built-into windows show that onFocus, the selection changes to all text.

I do not think this is a big enough deal to make it an option. We don't need more API surface area. Instead, we should just pick a behavior. 

I vote for whatever is the most simple to implement.

**Comment by BDisp** [comment]
> Is that a Visual Studio dialog?

Yes.


> I vote for whatever is the most simple to implement.

The most simple is what is right now, keeping the text in the same state that was when leaving. Nonetheless, this behavior only work like this if the navigation is trough the keyboard. Set focus by mouse deselects the text and it's positioned on point where was clicked. So, I'll add the behavior of selecting all text if the focus is by mouse and keeping the current behavior for the keyboard. If you don't agree please say, thanks.

**Comment by BDisp** [comment]
Fixed by #4554.
